react/src/components/BookList/index.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import BookItem from './components/BookItem';
+import {data} from '../../mock.js';
+
+class BookList extends Component {
+  render() {
+    const filterText = this.props.filterText;
+    const filterField = this.props.filterField;
+    let books = data.map(function(book, thisArg){
+      if(
+        !filterText ||
+        !filterField ||
+        (filterField === "1" && book.title.toLowerCase().includes(filterText.toLowerCase())) ||
+        (filterField === "2" && book.author.toLowerCase().includes(filterText.toLowerCase()))
+      ){
+        return (<BookItem key={book.id}
+          id={book.id}
+          author={book.author}
+          title={book.title}
```

Comment:
We could simplify this by writing:
```
return <BookItem key={book.id} {...book} />;
```
✨

Reply:
The `...` is called `spread operator`.
More magical stuff here:
https://davidwalsh.name/spread-operator